E-periodica Switzerland.js

```diff
@@ -0,0 +1,343 @@
+{
+	"translatorID": "dbfd99e3-6925-4b71-92b8-12b02aa875fc",
+	"label": "E-periodica Switzerland",
+	"creator": "Alain Borel",
+	"target": "^https?://(www|news?)\\.e-periodica\\.ch",
+	"minVersion": "5.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2023-07-23 10:37:03"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2023 Alain Borel
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if (url.includes('/digbib/view')) {
+		return "journalArticle";
+	}
+	else if (url.includes('/digbib/doasearch') && getSearchResults(doc, true)) {
+		return "multiple";
+	}
+	else {
+		return false;
+	}
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('h2.ep-result__title > a');
+	for (let row of rows) {
+		// Zotero.debug(row.innerHTML);
+		let href = row.href;
+		// Zotero.debug(href);
+		let title = ZU.trimInternal(row.textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+async function doWeb(doc, url) {
+	if (detectWeb(doc, url) == 'journalArticle') {
+		await scrape(doc, url);
+	}
+	else if (detectWeb(doc, url) == 'multiple') {
+		let items = await Zotero.selectItems(getSearchResults(doc, false));
+		if (!items) return;
+		for (let url of Object.keys(items)) {
+			await scrape(await requestDocument(url));
+		}
+	}
+	else {
+		// The fallback is not expected to be used on E-periodica, but just in case...
+		await scrape(doc, url);
+	}
+}
+
+async function scrape(nextDoc, url) {
+	var nextUrl = nextDoc.location.href;
+	//Zotero.debug('trying to process ' + nextUrl);
+	// Do we really need to handle these #-containing URLs?
+	nextUrl = nextUrl.replace("#", "%3A%3A").replace("::", "%3A%3A");
+	nextUrl = nextUrl.split("%3A%3A");
+	nextUrl = nextUrl[0] + "%3A%3A" + nextUrl.slice(-1);
```

Comment:
The safety and clarity of a regexp is very much in the eye of the beholder, in my opinion ;-) But I agree this section will be easier to maintain if it's better written and documented. I will try to follow your recommendation, and update the comments to provide answers instead of obsolete questions.

Reply:
I needed some more time to better understand your suggestion, I can now agree that is both safer than my original code (I was pretty sure about that already) + definitely easier to explain in the comments. I just had to use `articleURL.hash.replace(/^#/, "")` instead of `articleURL.hash.replace("^#", "")`.